Misc updates in gammapy.spectrum

joleroi opened the pull request:
This PR is a collection of smaller updates I made to ``gammapy.spectrum`` last week.

* Add  ``__sub__`` to ``CountsSpectrum``
* Add ``plot_hist`` method to ``CountsSpectrum``
* Make units optional when plotting IRFs 
* Make ``off_vector`` optional in ``SpectrumObservation``
* Misc updates to ``spectrum.results`` mostly plotting

joleroi commented:
@cdeil ok to merge?

cdeil commented on gammapy/spectrum/results.py:
```diff
@@ -316,6 +345,61 @@ def plot_fit(self):
+
+        Parameters
+        ----------
+        ax : `~matplolib.axes`, optional
```
Fix everywhere:
```
`~matplotlib.axes.Axes`
```

cdeil commented:
I left two inline comments.
Otherwise: add a test that exercises each new method?

cdeil commented:
Since you're doing misc updates to `gammapy.spectrum`: I recently noticed this:
https://github.com/gammapy/gammapy/blob/master/gammapy/spectrum/tests/test_counts_spectrum.py#L50
Fix that test here?

joleroi commented on gammapy/spectrum/spectrum_fit.py:
```diff
@@ -210,15 +199,17 @@ def _run_sherpa_fit(self):
-        log.info(str(self))
+        from sherpa.utils.err import IdentifierErr
+        
+        # TODO: Make this optional
```
You say I should not mute the logger from here, right?

The problem is that, since sherpa is imported in this function (and typically not in a user script) I only have access to the logger from here. I muted it because it is annoying when fitting O(100) simulated spectra. Do you propose to leave the logger as is and make users 'hack' it away (by importing sherpa) themselves?

cdeil replied:
> You say I should not mute the logger from here, right?

Yes.

> The problem is that, since sherpa is imported in this function (and typically not in a user script) I only have access to the logger from here. I muted it because it is annoying when fitting O(100) simulated spectra. Do you propose to leave the logger as is and make users 'hack' it away (by importing sherpa) themselves?

Yes, why not let users configure the Sherpa logger from their script?

If it helps, you can expose a convenience option for that from `gammapy.spectrum`.
But I think all that's needed here is some examples or two sentences in the docs to tell people how to configure the Sherpa logger?

joleroi commented:
Thanks for the review!

> Fix that test here?

I removed this test since the ``npred`` calculation functionality is already tested [here](
https://github.com/gammapy/gammapy/blob/master/gammapy/spectrum/tests/test_spectrum_fit.py#L58)

I added test for each new function, so I will merge as soon as checks pass
